python-package/lightgbm/sklearn.py

```diff
@@ -1169,6 +1217,19 @@ def feature_names_in_(self) -> np.ndarray:
 class LGBMRegressor(_LGBMRegressorBase, LGBMModel):
     """LightGBM regressor."""
 
+    def _more_tags(self) -> Dict[str, Any]:
+        # handle the case where ClassifierMixin possibly provides _more_tags()
+        if callable(getattr(_LGBMRegressorBase, "_more_tags", None)):
+            tags = _LGBMRegressorBase._more_tags(self)
+        else:
+            tags = {}
+        # override those with LightGBM-specific preferences
+        tags.update(LGBMModel._more_tags(self))
+        return tags
+
+    def __sklearn_tags__(self) -> Optional["_sklearn_Tags"]:
+        return LGBMModel.__sklearn_tags__(self)
```

Comment:
We need `__sklearn_tags__()` in `LGBMRegressor` and `LGBMClassifier` due to MRO again, right?

Reply:
Yeah exactly. Because the `scikit-learn` mixin classes come first (https://github.com/microsoft/LightGBM/pull/6651#discussion_r1772666260), if/when they define a `__sklearn_tags__()`, it would take precedence over the one from `LGBMModel`.

Reply:
`__sklearn_tags__` doesn't have any corresponding manual MRO walk the same way that `_more_tags` did. You should treat it like any other normal overriding of a method in python's OOP model.

Reply:
I don't understand what you mean by this comment. Would appreciate your thoughts on https://github.com/microsoft/LightGBM/pull/6651#discussion_r1777859058 if you have time.